### DISCUSSION TITLE
Uploading a file in a cron job

### DISCUSSION THREAD

**Discussion by ettisberger** [strapi/strapi#8714] [Legacy Topics]
Hey!

Im trying to upload a file in my cron job in the strapi backend. 

What Im trying to do:

- creating a pdf with pdfkit and get a buffer of it
- creating a new model entry with strapi services
- adding the pdf as a new file to the model

```
const newEntry = await strapi.query('mymodel').create({
   field1,
   field2
});

const pdf = fs.readFileSync('config/something/mynew.pdf');

// how to add the file as param?
const files = [??????];

await strapi.entityService.uploadFiles(newEntry,
    files, 
    {
      model: 'mymodel',
    }
);
```

I also tried it with using with

`strapi.plugins['upload'].services.upload.upload`

but I just cant figure out how I need to add the `files` parameter to make it work. Can someone give me some directions?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

**Reply by ettisberger** [reply]
thanks, I wrote my problem there and will check if I get some help :) Thanks again (we already talked once on slack).
